src/pages/Overall.js

```diff
@@ -0,0 +1,145 @@
+import React, { Component } from 'react'
+import "./Overall.scss"
+import OverallData from '../components/OverallData'
+import regions from "../data/regions.json"
+import "./Overall-Mobile.scss"
+
+export default class Data extends Component {
+    constructor(props) {
+        super(props)
+        this.state = {
+            regions: regions,
+            searchedRegions: regions,
+            filterType: "region",
+            isDesktop: false
+        }
+        this.updatePredicate = this.updatePredicate.bind(this)
+
+    }
+
+    componentDidMount() {
+        this.updatePredicate();
+        window.addEventListener("resize", this.updatePredicate);
+    }
+
+    updatePredicate() {
+        this.setState({ isDesktop: window.innerWidth > 750 });
+    }
+
+    changeHandler = (e) => {
+        const filter = e.target.value
+        this.setState({ filterType: filter })
+    }
+
+    searchHandler = (e) => {
+        var searchQuery = e.target.value
+
+        if (this.state.filterType === "region") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.region.toLowerCase().indexOf(searchQuery.toLowerCase()) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        } else if (this.state.filterType === "lastinput") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.lastinput.toString().indexOf(searchQuery.toString()) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        }
+        else if (this.state.filterType === "formnumbers") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.formnumbers.toString().indexOf(searchQuery) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        } else if (this.state.filterType === "voternumbers") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.voternumbers.toString().indexOf(searchQuery) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
+        } else if (this.state.filterType === "update") {
+            var searchResult = this.state.regions.filter((region) =>
+                region.update.toString().indexOf(searchQuery) !== -1
+            )
+            this.setState({ searchedRegions: searchResult })
```

Comment:
Every block as this line `this.setState({ searchedRegions: searchResult })`, why not hoist it out?